hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBCheckpointManager.java

```diff
@@ -85,6 +87,8 @@ public RocksDBCheckpoint createCheckpoint(String parentDir, String name) {
       LOG.info("Created checkpoint at {} in {} milliseconds",
               checkpointPath, duration);
 
+      waitForCheckpointDirectoryExist(checkpointPath.toFile());
+
       return new RocksDBCheckpoint(
           checkpointPath,
           currentTime,
```

Comment:
I believe checkpoint object returned should still be valid if the rocksDB api returned success. However, we should throw the timeout exception all the way back to the application. Application can wait for the snapshot dir creation when this happens before using the snapshot dir. The exception thrown should contain enough details so that application can wait on the dir creation.

Reply:
As discussed offline, it is not good idea to throw exception in case of timeout. We have to look into more details how we can make sure that rocksDB flush the queue and create checkpointing dir.